docs/src/overview/build-deploy.md

```diff
@@ -1,118 +1,83 @@
+
+This means the build is independent of the given environment and development environments are perfect copies of production.
+
+This independence does mean you can't connect to services (like databases) during the build.
+Once the app has gone through all of the build steps, it can connect to services in the deploy process.
+
+### Build steps
+
+1. **Validate configuration**:
+   The configuration is checked by validating the `.platform` directory and scanning the repository for any app configurations to validate individually.
+1. **Pull container images**:
+   Any container images that have been built before and that don't have any changes are pulled to be reused.
+1. **Install dependencies**:
+   If you have specified additional global dependencies, they're downloaded during this step.
+   This is useful for commands you may need in the build hook.
+1. **Run build flavor commands**:
+   For some languages (NodeJS, PHP), a series of standard commands are run based on the build flavor.
+   You can change the flavor or skip the commands by specifying it in your app configuration file.
+1. **Run build hook**:
+   The `build` hook comprises one or more shell commands that you write to finish creating your production code base.
+   It could be compiling Sass files, running a bundler, rearranging files on disk, or compiling.
+   The committed build hook runs in the build container.
+   During this time, commands have write access to the file system, but there aren't connections to other containers (services and other apps).
+1. **Freeze app container**:
+   The file system is frozen and produces a read-only container image, which is the final build artifact.
+
+## The deploy
+
+The deploy process connects each container from the build process and any services.
+The connections are defined in your app and services configuration.
+
+So unlike the build process, you can now access other containers,
+but the file system is read-only.
+
+### Deploy steps
+
+1. **Hold requests**:
+   Incoming requests are held to prevent service interruption.
+1. **Unmount current containers**:
+   Any previous containers are disconnected from their file system mounts.
+1. **Mount file systems**:
+   The file system is connected to the new containers.
+   New branches have file systems cloned from their parent.
+1. **Expose services**:
+   Networking connections are opened between any containers specified in your app and services configurations.
+1. **Run start commands**:
+   The commands necessary to start your app are run.
+1. **Run deploy hook**:
+   The `deploy` hook is any number of shell commands you can run to finish your deployment.
+   This can include clearing caches, running database migrations, and setting configuration that requires relationship information.
+1. **Serve requests**:
+  Incoming requests to your newly deployed application are allowed.
+
+After the deploy process is over, any commands in your `post_deploy` hook are run.
```

Comment:
Do we want to add a next reads on these topics here at the bottom or connect the build and deploy hook to the docs on those?

Reply:
I added a couple of links here and a link in the hooks doc to here (in #2017)

Reply:
Looks great now, thanks!